### PR TITLE
Replace from env_logger to tracing

### PR DESCRIPTION
**Pull request by MeenuyD**
refactor: switch logging from env_logger to tracing

The `wash` codebase historically used env_logger for logging. This commit refactors the code to utilize the tracing framework, aligning it with the logging strategy employed by other crates in the mono repo.

This change enhances consistency across the mono repo.
#994

### CODE REVIEW

**Comment by vados-cosmonic** [comment]
Hey @MeenuyD thanks for taking a shot at this! Before we get into the code review, would you mind getting set up with our DCO and signing your commits?  GitHub has a [helpful guide here on the check failure](https://github.com/wasmCloud/wasmCloud/pull/997/checks?check_run_id=18595298929) 

There's also [our `CONTRIBUTING.md` which includes information about the DCO and why we use it as a CNCF project](https://github.com/wasmCloud/wasmCloud/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

File: Cargo.toml [hunk 1]
```diff
@@ -95,7 +95,7 @@ console = { version = "0.15", default-features = false }
 data-encoding = { version = "2.3", default-features = false }
 dialoguer = { version = "0.10", default-features = false }
 dirs = { version = "4", default-features = false }
-env_logger = { version = "0.10", default-features = false }
+tracing = { version = "0.10", default-features = false }
```
Comment (connorsmith256) [inline Cargo.toml:98]: It looks like v0.1.40 is the latest? https://crates.io/crates/tracing

File: crates/wash-cli/src/bin/wash.rs [hunk 2]
```diff
@@ -220,7 +220,7 @@ enum CliCommand {
 #[tokio::main]
 async fn main() {
     use clap::CommandFactory;
-    env_logger::init();
+    tracing::init();
```
Comment (connorsmith256) [inline crates/wash-cli/src/bin/wash.rs:223]: I don't think this is a function in `tracing`

**Review by connorsmith256 — CHANGES_REQUESTED** [review]

File: crates/wash-cli/src/bin/wash.rs (hunk 2 above)
Comment (thomastaylor312) [reply crates/wash-cli/src/bin/wash.rs:223]: Yep as a heads up it will be in `tracing_subscriber`

File: Cargo.toml [hunk 3]
```diff
@@ -95,7 +95,7 @@ console = { version = "0.15", default-features = false }
 data-encoding = { version = "2.3", default-features = false }
 dialoguer = { version = "0.10", default-features = false }
 dirs = { version = "4", default-features = false }
-env_logger = { version = "0.10", default-features = false }
+tracing = { version = "0.1.40", default-features = false }
```
Comment (rvolosatovs) [inline Cargo.toml:98]:
```suggestion
tracing = { version = "0.1", default-features = false }
```

File: Cargo.toml [hunk 4]
```diff
@@ -158,7 +158,7 @@ tokio-util = { version = "0.7", default-features = false }
 toml = { version = "0.7", default-features = false }
 tracing = { version = "0.1", default-features = false }
 tracing-futures = { version = "0.2", default-features = false }
-tracing-opentelemetry = { version = "0.20", default-features = false }
+tracing-opentelemetry = { version = "0.22.0", default-features = false }
```
Comment (rvolosatovs) [inline Cargo.toml:161]:
```suggestion
tracing-opentelemetry = { version = "0.22", default-features = false }
```

File: crates/providers/Cargo.toml [hunk 5]
```diff
@@ -41,7 +41,7 @@ serde_json = { version = "1", default-features = false }
 thiserror = { version = "1", default-features = false }
 tokio = { version = "1", default-features = false }
 toml = { version = "0.8", default-features = false }
-tracing = { version = "0.1", default-features = false }
+tracing = { version = "0.1.40", default-features = false }
```
Comment (rvolosatovs) [inline crates/providers/Cargo.toml:44]:
```suggestion
tracing = { version = "0.1", default-features = false }
```

File: vendor/capability-providers/sqldb-postgres/Cargo.toml [hunk 6]
```diff
@@ -25,7 +25,7 @@ bb8-postgres = "0.8.1"
 bit-vec = "0.6"
 uuid = "1.1.2"
 toml = "0.5"
-tracing = "0.1"
+tracing = "0.1.40"
```
Comment (rvolosatovs) [inline vendor/capability-providers/sqldb-postgres/Cargo.toml:28]:
```suggestion
tracing = "0.1"
```

File: vendor/capability-providers/messaging-kafka/Cargo.toml [hunk 7]
```diff
@@ -8,7 +8,7 @@ resolver = "2"
 async-trait = "0.1"
 futures = "0.3"
 serde = {version = "1.0", features = ["derive"] }
-tracing = "0.1.37"
+tracing = "0.1.40"
```
Comment (rvolosatovs) [inline vendor/capability-providers/messaging-kafka/Cargo.toml:11]:
```suggestion
tracing = "0.1.37"
```

**Comment by brooksmtownsend** [comment]
Hey @MeenuyD I'm going to close this one as it's gone a bit out of date from `main`, let's discuss further in #994 if necessary

**Pull request closed without merging**